source/hpack.c

```diff
@@ -538,6 +524,105 @@ int aws_hpack_resize_dynamic_table(struct aws_hpack_context *context, size_t new
     return AWS_OP_SUCCESS;
 }
 
+int aws_hpack_insert_header(struct aws_hpack_context *context, const struct aws_http_header *header) {
+
+    /* Don't move forward if no elements allowed in the dynamic table */
+    if (AWS_UNLIKELY(context->dynamic_table.max_size == 0)) {
+        return AWS_OP_SUCCESS;
+    }
+
+    const size_t header_size = aws_hpack_get_header_size(header);
+
+    /* If for whatever reason this new header is bigger than the total table size, burn everything to the ground. */
+    if (AWS_UNLIKELY(header_size > context->dynamic_table.max_size)) {
+        return s_dynamic_table_shrink(context, 0);
```

Comment:
If the other side claims to have fit this into their dynamic table, which should be impossible, it seems like we're in undefined behavior town. Raise an error and fail the connection?